src/Tools/BuildBoss/ProjectCheckerUtil.cs

```diff
@@ -236,6 +237,36 @@ private bool CheckPackageReferences(TextWriter textWriter)
             return allGood;
         }
 
+        private bool CheckInternalsVisibleTo(TextWriter textWriter)
+        {
+            var allGood = true;
+            foreach (var internalsVisibleTo in _projectUtil.GetInternalsVisibleTo())
+            {
+                if (string.Equals(internalsVisibleTo.LoadsWithinVisualStudio, "false", StringComparison.OrdinalIgnoreCase))
+                {
+                    // IVTs explicitly declared with LoadsWithinVisualStudio="false" are allowed
+                    continue;
+                }
+
+                if (_projectUtil.Key.FileName.StartsWith("Microsoft.CodeAnalysis.ExternalAccess."))
+                {
+                    // External access layer may have external IVTs
+                    continue;
+                }
+
+                var builtByThisRepository = _solutionMap.Keys.Any(projectKey =>
+                    projectKey.FileName == $"{internalsVisibleTo.TargetAssembly}.csproj"
+                    || projectKey.FileName == $"{internalsVisibleTo.TargetAssembly}.vbproj");
```

Comment:
Nit: Really wish the || was on the previous line so the two conditions align so it's easier to see it's just the extension changing.

Reply:
I had to change this in f36a33cae3b98094508b46ea4545307a4c1a566a